game.py

```diff
@@ -0,0 +1,150 @@
+import random
+import string
+
+
+class Game:
+    def __init__(self, size=3, target=100, level='medium'):
+        self.size = size
+        self.target = target
+        self.level = level
+        self.characters = string.ascii_letters
+        self.board = [[None] * self.size for _ in range(size)]
+
+    def set_up_game(self):
+        self.set_mode()
+        self.set_values()
+        self.score = 0
+        self.winner = False
+        self.set_up_board()
+
+    def set_mode(self):
+        if self.level == 'easy':
+            self.characters = self.characters[:self.size - self.size // 2]
+        elif self.level == 'medium':
+            self.characters = self.characters[:self.size - self.size // 3]
+        elif self.level == 'hard':
+            self.characters = self.characters[:self.size]
+
+    def set_values(self):
+        self.delete = set()
+        self.create_items = []
+
+    def play_game(self, row, column):
+        if self.winner:
+            return
+        self.set_values()
+        self.make_move(row, column)
+        if self.no_next_move():
+            return 'no more move'
+
+    def set_up_board(self):
+        for row in range(self.size):
+            for column in range(self.size):
+                character = random.choice(self.characters)
+                self.board[row][column] = character
+        if self.no_next_move():
+            self.set_up_board()
+
+    def make_move(self, row, column):
+        self.check_move(row, column)
+        if len(self.delete) < 3:
+            return
+        self.delete_items()
+        self.replace_items()
+        self.create_new_items()
+        if self.score >= self.target:
+            self.winner = True
+
+    def check_move(self, row, column):
```

Comment:
what is this checking? and it seems to be doing more than checking, it is changing state of the class

Reply:
right, it returns a list of indices where items will be deleted. I renamed it get_indices_items_to_delete